daffodil-lib/src/main/scala/edu/illinois/ncsa/daffodil/oolag/OOLAG.scala

```diff
@@ -131,18 +131,30 @@ object OOLAG extends Logging {
    * already. This insures that the value exists for 'foo', or any errors/warnings
    * to be determined by its calculation have been recorded.
    */
-  abstract class OOLAGHost private (oolagContextArg: OOLAGHost, nArgs: Args)
+  abstract class OOLAGHostImpl private (
```

Comment:
Also sclaadoc should explain this dual track thing where we have OOLAGHost trait, and OOLAGHostImpl which is a class.

We needed this to enable richer combinations of mixins.